torch_geometric/data/storage.py

```diff
@@ -734,6 +734,65 @@ def is_edge_attr(self, key: str) -> bool:
             self._cached_attr[AttrType.NODE].add(key)
             return False
 
+    def _index_select(self, keys: List[str], dims: List[int], index: Tensor):
+        assert len(keys) == len(
+            dims
+        ), "Number of provided keys does not match number of provided dims."
+        for key, dim in zip(keys, dims):
+            if key in self:
+                self[key] = torch.index_select(self[key], dim, index)
+
+    def _dims_for_attrs(self, keys: List[str]) -> List[int]:
+        # to avoid errors when 'num_edges == num_edge_features' or
+        # 'num_nodes == num_node_features'
+        predefined_dims = dict(x=0, edge_index=1, edge_attr=0)
+        dims = []
+        for key in keys:
+            if key in predefined_dims.keys():
+                dim = predefined_dims[key]
+            elif self.is_edge_attr(key):
+                dim = (torch.tensor(
+                    self[key].size()) == self.num_edges).nonzero()[0].item()
+            elif self.is_node_attr(key):
+                dim = (torch.tensor(
+                    self[key].size()) == self.num_nodes).nonzero()[0].item()
+            dims.append(dim)
+        return dims
+
+    def is_sorted_by_time(self) -> bool:
+        if 'time' in self:
+            return bool(torch.all(self.time[:-1] <= self.time[1:]))
+        return True
+
+    def sort_by_time(self) -> 'GlobalStorage':
+        if self.is_sorted_by_time():
+            return self
+        if 'time' in self:
+            self.time, perm = torch.sort(self.time, stable=True)
+            if self.is_edge_attr('time'):
+                keys = self.edge_attrs()
+                keys.remove('time')
+                self._index_select(keys, self._dims_for_attrs(keys), perm)
+            # TODO: cover case where 'time' is a node attribute
+        return self
+
+    def snapshot(self, start_time: Union[float, int],
```

Comment:
Hi @DamianSzwichtenberg, I think the `snapshot` function should accept the `start`-th and `end`-th time steps, as people usually don't care about the exact time values. For example, one can call `snapshot` to obtain its graph snapshot from 0th to 2rd time step: `snapshot(0, 2)`

Reply:
Hi @EdisonLeeeee, thanks for the feedback. As `TemporalData` was representing CTDGs, I assumed that the same applies here - hence user will not operate on `start`-th and `end`-th steps, but on concrete `time` values. In my understanding `snapshot` may be used to generate a DTDGs. Please correct me if I'm wrong.